src/ReduxConfirmDialog/reduxConfirmDialogReducer.ts

```diff
@@ -34,41 +34,38 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { Reducer } from 'react';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 
 import { ConfirmDialogState, RootState } from '../state';
 
-type ACTIONS = 'SHOW_REDUX_CONFIRM_DIALOG' | 'HIDE_REDUX_CONFIRM_DIALOG';
+const initialState: ConfirmDialogState = {};
 
-const SHOW_REDUX_CONFIRM_DIALOG = 'SHOW_REDUX_CONFIRM_DIALOG';
-export const showReduxConfirmDialogAction = ({
-    ...args
-}: ConfirmDialogState) => ({
-    type: SHOW_REDUX_CONFIRM_DIALOG,
-    ...args,
-});
-const HIDE_REDUX_CONFIRM_DIALOG = 'HIDE_REDUX_CONFIRM_DIALOG';
-export const hideReduxConfirmDialogAction = () => ({
-    type: HIDE_REDUX_CONFIRM_DIALOG,
+const slice = createSlice({
+    name: 'reduxConfirmDialog',
+    initialState,
+    reducers: {
+        showReduxConfirmDialogAction: (
+            state,
+            action: PayloadAction<ConfirmDialogState>
+        ) => {
+            state = { ...state, ...action };
+        },
+        hideReduxConfirmDialogAction: state => {
+            state.callback = undefined;
+            state.title = undefined;
+            state.content = undefined;
+            state.confirmLabel = undefined;
+            state.cancelLabel = undefined;
+            state.onOptional = undefined;
+            state.optionalLabel = undefined;
```

Comment:
If you want to [reset the whole state](https://redux-toolkit.js.org/usage/immer-reducers#resetting-and-replacing-state) back to the initial one, you can return the new state instead. Like this:
```suggestion
            return initialState;
```

Reply:
Nice, i didnt realize it was possible to just return the new state. I thought redux toolkit needed it to mutate the proxy state :)